internal/controller/pipelinerollout_controller.go

```diff
@@ -55,6 +57,18 @@ const (
 	numWorkers                = 16 // can consider making configurable
 )
 
+var pipelineROReconciler *PipelineRolloutReconciler
+
+const (
+	PipelinePhaseUnknown   = ""
+	PipelinePhaseRunning   = "Running"
+	PipelinePhaseSucceeded = "Succeeded"
+	PipelinePhaseFailed    = "Failed"
+	PipelinePhasePausing   = "Pausing"
+	PipelinePhasePaused    = "Paused"
+	PipelinePhaseDeleting  = "Deleting"
+)
+
```

Comment:
yeah, good question. I guess we need to decide where in our code we still want to reference numaflow directly and where we don't. Theoretically, if we expect something to not change in Numaflow, then we may be okay to reference it directly, right? And eliminate copy/pasting into our code.

Reply:
I'm also not sure. We should have a separate PR to address that by looking at the entire code base and see where the dependency is used and if it could either be removed or acceptable to stay.